## Задание 1.4. Работа c несколькими списками

Даны два списка: Дата покупки и Суммы покупок по датам

Данные:

<i>['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
[1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]</i>

### Найдите, какая выручка у компании в ноябре. Используйте list comprehensions.

In [7]:
# Подключаем модуль, который беспечивает поддержку выполнения аннотации типов
from typing import List 

# Создаем функцию для вычисления выручки
def calculate_revenue(purchase_dates: List[str], purchase_amounts: List[int]) -> int:
    # Получаем только даты из набора данных
    dates = [date.split('-')[1] for date in purchase_dates]
    
    # Используем zip для объединения дат и сумм
    combined_data = zip(dates, purchase_amounts)
    
    # Вычисляем выручку за ноябрь (11-ый месяц)
    november_revenue = sum(amount for date, amount in combined_data if date == '11')
    
    return november_revenue

# Записываем данные из условия задачи
purchase_dates = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
purchase_amounts = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

# Запускаем созданую функцию 
revenue = calculate_revenue(purchase_dates, purchase_amounts)

# Выводим на экран итоговую выручку за ноябрь 
print("Выручка компании в ноябре:", revenue)

Выручка компании в ноябре: 25098


### Найдите выручку компании в зависимости от месяца. Напишите функцию, которая на вход принимает списки с датами и с выручкой, а на выходе вернет словарь, где ключи - это месяцы, а значения - это выручка.

Примеры/Тесты:

<i><finction_name>(dates, prices) -> {'09': 25647, '12': 45452, '10': 28645, '11': 25098}</i>

In [12]:
# Подключаем модули, который беспечивает поддержку выполнения аннотации типов и работы со словарями
from typing import List, Dict

# Создаем функцию для вычисления выручки
def calculate_monthly_revenue(purchase_dates: List[str], revenue_amounts: List[int]) -> Dict[str, int]:
    # Создаем пустой словарь для хранения данных о ежемесячной выручке
    monthly_revenue = {}
    
    # Используем zip для объединения дат и сумм
    for date, revenue in zip(purchase_dates, revenue_amounts):
        # Получаем только цифры месяца из набора данных
        month = date.split('-')[1]
        
        # Проверяем существование данных для конкретного месяца 
        if month in monthly_revenue:
            # Если данные существуют, то добавляем доход за месяц в выручку 
            monthly_revenue[month] += revenue
        else:
            # Если данных ранее нет, то устанавливаем новое значение как доход за месяц
            monthly_revenue[month] = revenue
    
    return monthly_revenue

# Записываем данные из условия задачи
purchase_dates = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
revenue_amounts = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

# Запускаем созданую функцию 
monthly_revenue = calculate_monthly_revenue(purchase_dates, revenue_amounts)
print("Выручка компании по месяцам:", monthly_revenue)



Выручка компании по месяцам: {'09': 25647, '12': 45452, '10': 28645, '11': 25098}
